src/components/connections.vue

```diff
@@ -14,14 +14,39 @@ export default {
     }
   },
   methods: {
-    ...mapActions(['connect', 'setDatabase', 'setCollection']),
+    ...mapActions([
+      'connect',
+      'setDatabase',
+      'setCollection',
+      'lastOpenedCollection',
+      'addCollectionToOpenedCollections'
+    ]),
     async doConnect() {
       await this.connect(this.connectionString)
       this.$router.push('/connection')
+    },
+    setSelectedDBorCollection(sourceType = 'db', sourceName) {
+      if (sourceType === 'db') {
+        if (this.databaseName == sourceName) {
+          return false
+        }
+
+        this.setDatabase(sourceName)
+      } else {
+        if (this.collectionName == sourceName) {
+          return false
+        }
+
+        if (this.openedCollections.includes(sourceName)) {
+          this.setCollection(sourceName)
+        } else {
+          this.addCollectionToOpenedCollections(sourceName)
```

Comment:
this entire logic should reside in the store, in `setCollection`. the component shouldn't care about the differences of `setCollection` and `addCollectionToOpenedCollections`. it should always call one method, and that method should choose to `addCollectionToOpenedCollections` if it's opened for the first time.